spring-boot/filemanager-spring-boot-autoconfigure/src/main/java/com/opuscapita/filemanager/configuration/FileManagerConfigurationProperties.java

```diff
@@ -0,0 +1,32 @@
+package com.opuscapita.filemanager.configuration;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+@Data
+@AllArgsConstructor
+@RequiredArgsConstructor
+@ConfigurationProperties(prefix = "filemanager")
+public class FileManagerConfigurationProperties {
+
+    private WebProperties web;
+
+    private FileSystemProperties filesystem;
+
+    @Data
+    @AllArgsConstructor
+    @RequiredArgsConstructor
+    public static class FileSystemProperties {
+        private String rootPath;
+        private String rootName;
+    }
+
+    @Data
+    @AllArgsConstructor
+    @RequiredArgsConstructor
```

Comment:
Will this work without these 3 @RequiredArgsConstructor annotations? @Data includes @RequiredArgsConstructor

Reply:
I would replace `@RequiredArgsConstructor` with `@NoArgsConstructor`. Makes more sense to me since there are no required variables anyway.